pkg/runtime/wfengine/workflow.go

```diff
@@ -322,13 +322,31 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 
 	// The logic/for loop below purges/removes any leftover state from a completed or failed activity
 	// TODO: for optimization make multiple go routines and run them in parallel
-	for _, e := range state.Inbox {
+	for num, e := range state.Inbox {
 		var taskID int32
 		if ts := e.GetTaskCompleted(); ts != nil {
 			taskID = ts.TaskScheduledId
 		} else if tf := e.GetTaskFailed(); tf != nil {
 			taskID = tf.TaskScheduledId
 		} else {
+			if strings.Contains(e.String(), "timerFired") {
+				numZero := 6 - len(fmt.Sprint(num))
```

Comment:
To avoid expensive conversions to string
```suggestion
				numZero := 5 - int(math.Log10(float64(num)))
```

Reply:
Please add a constant for the magic number `5` or `6`

Reply:
qq Why are we adding a `"timer"` record here after the `"timer"` has been fired?